### PR TITLE
fix(shell/server): remove unnecessary dependencies of info_task and use intermediate envs

### PR DESCRIPTION
**Pull request by Kha**
@gebner Looks good?

### CODE REVIEW

File: src/shell/server.cpp [hunk 1]
```diff
@@ -500,9 +491,10 @@ class server::info_task : public task<unit> {
                 auto opts = m_server->m_ios.get_options();
                 auto env = m_server->m_initial_env;
                 if (auto mod = m_mod_info->m_result.peek()) {
-                    if (mod->m_loaded_module->m_env)
-                        env = mod->m_loaded_module->m_env.get();
-                    if (!mod->m_snapshots.empty()) opts = mod->m_snapshots.back()->m_options;
+                    if (!mod->m_snapshots.empty()) {
```
Comment (Kha) [inline src/shell/server.cpp]: I suppose there's no way for the snapshots to be empty here

Comment (gebner) [reply src/shell/server.cpp]: Only if we disable snapshots completely.

Comment (Kha) [reply src/shell/server.cpp]: Right, in that case we should fall back to the completed environment if available.

**Comment by Kha** [comment]
I've always wondered why asynchronicity didn't break the interactive tests... makes sense if `info` has been basically synchronous up to now. We should probably add a flag `blocking` to `sync` and use it in interactive tests.

**Comment by gebner** [comment]
> I've always wondered why asynchronicity didn't break the interactive tests... makes sense if info has been basically synchronous up to now. We should probably add a flag blocking to sync and use it in interactive tests.

Because we don't do anything asynchronously in the tests. :smile:   We use the `-j0` flag in the tests, which disables all multi-threading (and potential non-determinism).

**Comment by Kha** [comment]
Huh, I guess I should actually look at the failed test before thinking about how to fix it :smile: . I now use `get_closest_snapshot` so that infos are printed in the correct local context (at least eventually, the `break_at_pos` parser may overtake the actual one).

**Pull request closed without merging**